src/contract_client/basic_node_signer.ts

```diff
@@ -1,21 +1,33 @@
-import { Keypair, TransactionBuilder, hash } from '..'
+import { Keypair, TransactionBuilder, hash } from "..";
+import type { AssembledTransaction } from "./assembled_transaction";
+import type { ContractClient } from "./client";
 
 /**
  * For use with {@link ContractClient} and {@link AssembledTransaction}.
  * Implements `signTransaction` and `signAuthEntry` with signatures expected by
  * those classes. This is useful for testing and maybe some simple Node
  * applications. Feel free to use this as a starting point for your own
  * Wallet/TransactionSigner implementation.
+ * @param keypair {@link Keypair} to use to sign the transaction or auth entry
+ * @param networkPassphrase passphrase of network to sign for
  */
-export const basicNodeSigner = (keypair: Keypair, networkPassphrase: string) => ({
+export const basicNodeSigner = (
+  /**
+   *{@link Keypair} to use to sign the transaction or auth entry
+   */
+  keypair: Keypair,
+  /**
+   * passphrase of network to sign for
+   */
+  networkPassphrase: string,
```

Comment:
I'm using both jsdoc syntax and TS syntax to document these params because I'm not sure which we should favor. I've more frequently used the TS version throughout, but our linter wants me to use jsdoc instead, and it doesn't care about the TS syntax.

Reply:
we use jsdoc almost everywhere (because reasons? iunno) and `yarn docs` might dgaf about ts syntax, so that's (unfortunately) preferred :pray:

Reply:
See [research](https://github.com/stellar/js-stellar-sdk/pull/962#discussion_r1596950321) – I think we should prefer TSDoc `/** inline param docs */` from now on!